### DISCUSSION TITLE
2021-07-27 planning discussion

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#490] [Maintainer discussions]
This week's planning discussion.

Topics:
- [New format](https://github.com/centerofci/mathesar/discussions/490#discussioncomment-1061880)
- [Active Work](https://github.com/centerofci/mathesar/discussions/490#discussioncomment-1061890)
- [Retrospective](https://github.com/centerofci/mathesar/discussions/490#discussioncomment-1061896)
- [Milestone deadline updates](https://github.com/centerofci/mathesar/discussions/490#discussioncomment-1061912)

Anyone is free to post other topics as well. Please update the list above (or ask a maintainer to) if you do.

@mathemancer, @eito-fis, @pavish your responses to all top-level posts are required. Everyone else is free to comment, regardless of level of involvement in the project.

Also an FYI that @ghislaineguerin is on vacation until Friday.

**Comment by kgodey** [comment]
## New format

The active work and retrospective portions of the update are now automatically generated and follow a new format. I'm going to resume posting on Tuesdays instead of Wednesdays since there's no data gathering step needed.

What do you think of the new formats and timing?

**Reply by mathemancer** [reply]
I definitely think the Tuesday is better.  I like that the script sets things up with links to the actual issues.  I think that'll make it easier to click through for more granular commenting on issues themselves.

**Reply by eito-fis** [reply]
Automatic setup and separate sections both seem great! Also +1 for links being nice.

**Reply by pavish** [reply]
I think this is much better!

**Comment by kgodey** [comment]
## Active Work

This post lists active work based on the [Active](https://github.com/centerofci/mathesar/projects/1) GitHub project. It also includes a current summary of milestones involved in active work.

### [2021-07-28] [03. Editable Tables](https://github.com/centerofci/mathesar/milestone/4)

- **[5 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2203.%20Editable%20Tables%22) already closed.** 
- **Issues currently in progress:**
  - [@pavish]  #196
- **Issues not yet started:**
  - [@pavish]  #195
  - [@pavish]  #197

### [2021-07-30] [2021-07 improvements](https://github.com/centerofci/mathesar/milestone/28)

- **[3 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%222021-07%20improvements%22) already closed.** 
- **Issues currently in progress:**
  - [@kgodey]  #485
  - [@pavish]  #358
  - [@andyharley-cci]  #225
  - [@Dhruvi16]  #322

### [2021-08-03] [04. Working with Schemas](https://github.com/centerofci/mathesar/milestone/6)

- **[8 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2204.%20Working%20with%20Schemas%22) already closed.** 
- **Issues currently in progress:**
  - [@eito-fis]  #365
  - [@eito-fis]  #437
- **Issues not yet started:**
  - [@dhruvkb]  #171
  - [@pavish]  #362
  - [@pavish]  #316

### [2021-08-09] [05. Tables from File Import](https://github.com/centerofci/mathesar/milestone/1)

- **[25 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2205.%20Tables%20from%20File%20Import%22) already closed.** 
- **Issues currently in progress:**
  - [@mathemancer]  #201
- **Issues not yet started:**
  - [@andyharley-cci]  #421
  - [@pavish]  #296
- **Issues remaining in backlog:**
  - 1 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2205.%20Tables%20from%20File%20Import%22) issues

### [2021-08-16] [06. Working with Tables](https://github.com/centerofci/mathesar/milestone/8)

- **[6 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2206.%20Working%20with%20Tables%22) already closed.** 
- **Issues currently in progress:**
  - [@eito-fis]  #190
  - [@kgodey]  #302
  - [@kgodey]  #326
  - [@ghislaineguerin]  #324
- **Issues not yet started:**
  - [@eito-fis]  #203
  - [@andyharley-cci]  #364
  - [@eito-fis]  #452
  - [@eito-fis]  #449
  - [@eito-fis]  #397
- **Issues remaining in backlog:**
  - 8 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2206.%20Working%20with%20Tables%22) issues

### [2021-08-23] [07. Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)

- **[2 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%2207.%20Initial%20Data%20Types%22) already closed.** 
- **Issues currently in progress:**
  - [@ghislaineguerin]  #238
  - [@ghislaineguerin]  #234
- **Issues not yet started:**
  - [@ghislaineguerin]  #239
  - [@ghislaineguerin]  #240
  - [@powellc]  #387
  - [@powellc]  #388
  - [@powellc]  #247
  - [@ghislaineguerin]  #235
  - [@ghislaineguerin]  #236
  - [@ghislaineguerin]  #237
  - [@ghislaineguerin]  #241
  - [@mathemancer]  #377
  - [@mathemancer]  #378
  - [@mathemancer]  #379
  - [@mathemancer]  #380
  - [@mathemancer]  #381
  - [@mathemancer]  #382
  - [@mathemancer]  #383
  - [@kgodey]  #367
  - [@kgodey]  #368
  - [@kgodey]  #369
- **Issues remaining in backlog:**
  - 39 [backend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20backend%22+milestone%3A%2207.%20Initial%20Data%20Types%22) and [database](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20database%22+milestone%3A%2207.%20Initial%20Data%20Types%22) issues
  - 8 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%2207.%20Initial%20Data%20Types%22) issues

### [2021-08-31] [2021-08 improvements](https://github.com/centerofci/mathesar/milestone/29)

- **[3 issues](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+is%3Aclosed+milestone%3A%222021-08%20improvements%22) already closed.** 
- **Issues currently in progress:**
  - [@mr-gabe49]  #483
- **Issues not yet started:**
  - [@mathemancer]  #280
  - [@patelaryan7751]  #290
- **Issues remaining in backlog:**
  - 8 [frontend](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20frontend%22+milestone%3A%222021-08%20improvements%22) issues
  - 6 [design](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20design%22+milestone%3A%222021-08%20improvements%22) issues
  - 2 [infrastructure](https://github.com/centerofci/mathesar/issues?q=is%3Aissue+project%3Acenterofci%2Fmathesar%2F2+is%3Aopen+label%3A%22work%3A%20infrastructure%22+milestone%3A%222021-08%20improvements%22) issues

## Questions
- Does this workload look okay to you?
- Is anything inaccurate?
- @pavish, @Dhruvi16, @andyharley-cci, @eito-fis, @dhruvkb: you have work in milestones due soon. Any blockers?

**Reply by mathemancer** [reply]
For me, this looks pretty good.  The type implementation issues will take some time (especially considering the ones still in the backlog), so it's good they're pushed back.  I'm not noticing any inaccuracies.

**Reply by eito-fis** [reply]
Workload seems reasonable to me, no real blockers either.

**Reply by pavish** [reply]
Workload looks okay, but tightly packed. There's little room for implementation hiccups/time to focus on discussions.

There were some hiccups in implementing record deletion which took to an extra day. I also have frontend interviews scheduled during the week. So, it might take a couple more days for Milestone 3.

Milestone 4, however, has very less time. It involves re-structuring layout and frontend code, so we will have to push the deadline.

**Reply by kgodey** [reply]
@pavish I updated the deadlines for milestone 3 to be Monday (Aug 2) and milestone 4 to be 1 week after that (Monday, Aug 9). This should give you enough time, hopefully. Milestone 5 (file import) is due on the same day as Milestone 4 now, but there's less work remaining there so hopefully that deadline works.

**Reply by pavish** [reply]
@kgodey Looks good to me.. Thanks!

**Comment by kgodey** [comment]
## Retrospective

This post lists issue updates from 2021-07-21 until now by milestone.

### [02. Basic Filtering, Sorting, Grouping](https://github.com/centerofci/mathesar/milestone/2)

- **Issues closed:**
  - [@kgodey]  #352
  - [@pavish]  #344
  - [@pavish]  #343
  - [@pavish]  #215

### [03. Editable Tables](https://github.com/centerofci/mathesar/milestone/4)

- **Issues currently in progress:**
  - [@pavish]  #196

### [2021-07 improvements](https://github.com/centerofci/mathesar/milestone/28)

- **Issues currently in progress:**
  - [@kgodey]  #485
  - [@pavish]  #358
  - [@andyharley-cci]  #225
  - [@Dhruvi16]  #322
- **New issues created:**
  -  #485, created by @pavish

### [04. Working with Schemas](https://github.com/centerofci/mathesar/milestone/6)

- **Issues currently in progress:**
  - [@eito-fis]  #365
  - [@eito-fis]  #437

### [05. Tables from File Import](https://github.com/centerofci/mathesar/milestone/1)

- **Issues currently in progress:**
  - [@mathemancer]  #201

### [06. Working with Tables](https://github.com/centerofci/mathesar/milestone/8)

- **Issues currently in progress:**
  - [@eito-fis]  #190
  - [@kgodey]  #302
  - [@kgodey]  #326
  - [@ghislaineguerin]  #324

### [07. Initial Data Types](https://github.com/centerofci/mathesar/milestone/11)

- **Issues closed:**
  - [@ghislaineguerin]  #229
- **Issues currently in progress:**
  - [@ghislaineguerin]  #238
  - [@ghislaineguerin]  #234

### [08. Working with Views](https://github.com/centerofci/mathesar/milestone/13)

- **New issues created:**
  -  #470, created by @kgodey
  -  #469, created by @kgodey
  -  #468, created by @kgodey
  -  #467, created by @kgodey
  -  #466, created by @kgodey
  -  #465, created by @kgodey
  -  #464, created by @kgodey

### [2021-08 improvements](https://github.com/centerofci/mathesar/milestone/29)

- **Issues closed:**
  - [@andyharley-cci]  #462
  - [@mr-gabe49]  #374
  - [@andyharley-cci]  #359
- **Issues currently in progress:**
  - [@mr-gabe49]  #483
- **New issues created:**
  -  #483, created by @pavish
  -  #474, created by @kgodey
  -  #473, created by @kgodey
  -  #462, created by @kgodey

### [2021-09 improvements](https://github.com/centerofci/mathesar/milestone/30)

- **New issues created:**
  -  #479, created by @pavish
  -  #475, created by @kgodey

## Questions
- What went well this week?
- What could have gone better?
- Any suggested process improvements?

**Reply by mathemancer** [reply]
I think implementing generally went well.  I'm glad we're hitting some deadlines and finishing some milestones.  As far as what could have gone better, I think I spent a bit of time going down the wrong path on the previews API work, trying to merge it with the views work.

I don't have any process improvements to suggest at the moment.

**Reply by pavish** [reply]
I think things went well this week. We were able to complete milestone 2 and also start work on frontend build. I am happy for frontend PRs from the community :) (Looking at you, @mr-gabe49).

I did not manage time and switching context effectively across development and discussions. I am focusing on dedicating fixed time for each. 

I do not have any suggestions at the moment.

**Reply by eito-fis** [reply]
I was out for this week, nothing much for me to add.

**Comment by kgodey** [comment]
# Milestone deadline updates

No major updates this week.
- [2021-07 improvements](https://github.com/centerofci/mathesar/milestone/28) is now due Jul 30 instead of Jul 31 so that it's due on a weekday.
-  [04. Working with Schemas](https://github.com/centerofci/mathesar/milestone/6) is now due Aug 3 instead of Aug 2 so that we don't have milestones due on consecutive workdays.
